sherpa/astro/data.py

```diff
@@ -1992,6 +1992,41 @@ def quality(self, val):
         #
         self._set_related("quality", val)
 
+    # It is unclear exactly what the quality_filter is meant to
+    # represent, so as part of improving support for ignore_bad, add
+    # some semantics.
+    #
+    @property
+    def quality_filter(self) -> Optional[np.ndarray]:
+        """The applied quality filter.
+
+        If set, this indicates which channels are currently selected
+        (a True value) and those that are ignored (False) by a
+        `ignore_bad` call.
+
+        Returns
+        -------
+        quality_filter : numpy.ndarray or None
+
+        See Also
+        --------
+        ignore_bad, quality
+
+        """
+        return self._quality_filter
+
+    # TODO: should this be allowed? Maybe this should only be set
+    # internally with ignore_bad?
+    #
+    @quality_filter.setter
+    def quality_filter(self, val: Optional[ArrayType]) -> None:
+        if val is None:
+            qfilt = None
+        else:
+            # ensure we store boolean values
+            qfilt = np.asarray(val, dtype=bool)
+        self._set_related("quality_filter", qfilt)
```

Comment:
Or maybe I just overlooked it. Either way, I would say that you can set the filter and the grouping by hand, but the quality filter is something that's inherent in the data. If you need to change that, change the quality array and reapply the filter. (Side note: Should be automatically do something, like reset or re-apply the quality filter when the quality array is changed?)
I just feel that there is no reasonable use case to set this filter directly and it's better to offer fewer overlapping setters (grouping, ignore/notice, and quality_filter all change which bins are used in fitting).

Reply:
Well, `quality_filter` is a field that already exists. We can make it read-only (and that would likely remove some of the complexities), but I think that is only really worth it if we improve the quality handling (which this PR is explicitly not about).

Reply:
Argh. Removing the setter and making the routines that need to change `self._quality_filter` directly fails because there's some code in `sherpa.astro.utils` that change the `quality_filter`. It is likely that this extra code can be re-written somehow, but that's not going to happen now, so I think we are stuck with this.

Reply:
So I've left it as is for now.